mesheryctl/pkg/utils/healthcheck.go

```diff
@@ -290,3 +290,138 @@ func WaitForPodRunning(c *meshkitkube.Client, desiredPod, namespace string, time
+			{
+				replicasets, err := c.KubeClient.AppsV1().ReplicaSets(namespace).List(context.TODO(), metav1.ListOptions{})
+				if err != nil {
+					return false, err
+				}
+				return len(replicasets.Items) == 0, nil
+			}
+
+		case "rolebinding":
+			{
+				rolebindings, err := c.KubeClient.RbacV1().RoleBindings(namespace).List(context.TODO(), metav1.ListOptions{})
+				if err != nil {
+					return false, err
+				}
+				return len(rolebindings.Items) == 0, nil
+			}
+
+		case "role":
+			{
+				roles, err := c.KubeClient.RbacV1().Roles(namespace).List(context.TODO(), metav1.ListOptions{})
+				if err != nil {
+					return false, err
+				}
+				return len(roles.Items) == 0, nil
+			}
+
+		case "statefulset":
+			{
+				statefulsets, err := c.KubeClient.AppsV1().StatefulSets(namespace).List(context.TODO(), metav1.ListOptions{})
+				if err != nil {
+					return false, err
+				}
+				return len(statefulsets.Items) == 0, nil
+			}
+
+		case "service":
+			{
+				services, err := c.KubeClient.CoreV1().Services(namespace).List(context.TODO(), metav1.ListOptions{})
+				if err != nil {
+					return false, err
+				}
+				return len(services.Items) == 0, nil
+			}
+		}
+
+		return true, nil
+	}
+}
+
+// Returns condition function to indicate that the `namespace` does not exist anymore.
+func isNamespaceDeleted(c *meshkitkube.Client, namespace string) wait.ConditionFunc {
+	return func() (bool, error) {
+		namespaces, err := c.KubeClient.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		// Check if namespace exists in namespaces list
+		for _, ns := range namespaces.Items {
+			if ns.Name == namespace {
+				return false, nil
+			}
+		}
+
+		return true, nil
+	}
+}
+
+// Poll up to timeout seconds until all the components have been deleted
+func pollForMesheryComponentsDeleted(c *meshkitkube.Client, namespace string, timeout time.Duration) error {
+	// Wait for pods to delete
+	err := wait.PollImmediate(time.Second, timeout, areMesheryComponentsDeleted(c, namespace, "pod"))
+	if err != nil {
+		return err
+	}
+	// Wait for replica sets to delete
+	err = wait.PollImmediate(time.Second, timeout, areMesheryComponentsDeleted(c, namespace, "replicaset"))
+	if err != nil {
+		return err
+	}
+	// Wait for role bindings to delete
+	err = wait.PollImmediate(time.Second, timeout, areMesheryComponentsDeleted(c, namespace, "rolebinding"))
+	if err != nil {
+		return err
+	}
+	// Wait for roles to delete
+	err = wait.PollImmediate(time.Second, timeout, areMesheryComponentsDeleted(c, namespace, "role"))
+	if err != nil {
+		return err
+	}
+	// Wait for stateful sets to delete
+	err = wait.PollImmediate(time.Second, timeout, areMesheryComponentsDeleted(c, namespace, "statefulset"))
+	if err != nil {
+		return err
+	}
+	// Wait for services to delete
+	err = wait.PollImmediate(time.Second, timeout, areMesheryComponentsDeleted(c, namespace, "service"))
+	if err != nil {
+		return err
+	}
+
```

Comment:
We shouldn't run multiple poll methods like this, it's bad practice.

Reply:
Oh, I did this so that it doesn't poll for everything everytime. For example if all pods are deleted and and some service is yet to be deleted, I would not want the pods to be polled again since we know they're deleted.

Reply:
See, what we should do is if any of the resources are present, we poll again. That's how it should work.

And this polling shouldn't call the boolean function each second on each failure instead, as we know stopping a system is a time taking process from the user's perspective and that's why we can take a lease here and call the function after an interval, maybe 7 Seconds.

makes sense??

Reply:
about the linter, it never works locally, at least for me. You can check your lint errors by clicking on 'Files Changed' in the top navigation.

Reply:
Yep got it, thanks! (also nudging you to check the comment I made under the issue :sweat_smile:)

Reply:
Yeah, I just checked it, that makes sense, You're right on that. I forgot the original issue and over-complexed it in my head and misdirected you.

I apologies! you can revoke these unnecessary changes. Let's get this wrapped up.

Reply:
That's completely fine! Thanks for sticking around for this for so long. Let's get this done with :rocket: